src/CBLURLEndpointListener_Internal.hh

```diff
@@ -0,0 +1,150 @@
+//
+//  CBLURLEndpointListener_Internal.hh
+//
+// Copyright © 2025 Couchbase. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#pragma once
+
+#include "Internal.hh"
+#include "CBLCollection_Internal.hh"
+#include "CBLDatabase_Internal.hh"
+#include "Defer.hh"
+#include "c4Listener.hh"
+
+#ifdef COUCHBASE_ENTERPRISE
+
+CBL_ASSUME_NONNULL_BEGIN
+
+struct CBLListenerAuthenticator {
+    
+};
+
+struct CBLURLEndpointListener final : public CBLRefCounted {
+public:
+    CBLURLEndpointListener(const CBLURLEndpointListenerConfiguration &conf)
+    : _conf(conf)
+    {
+        if (conf.collectionCount == 0) {
+            C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "No collections in CBLURLEndpointListenerConfiguration");
+        }
+    }
+
+    const CBLURLEndpointListenerConfiguration* configuration() const { return &_conf; }
+
+    uint16_t port() const {
+        if (_port == 0 && _c4listener) {
+            _port = _c4listener->port();
+        }
+        return _port;
+    }
+
+    FLMutableArray _cbl_nullable getUrls() const {
+        if (!_c4listener) return (FLMutableArray) nullptr;
+
+        CBLDatabase* cblDb = CBLCollection_Database(_conf.collections[0]);
```

Comment:
You can call _conf.collections[0]->database() to get the database object instead of using the C API.

Reply:
Sure